### DISCUSSION TITLE
Getting Started

### DISCUSSION THREAD

**Discussion by JuanCruz98** [material-components/material-web#4427] [Q&A]
Hi, so I was trying to get started following this: https://github.com/material-components/material-web/blob/main/docs/quick-start.md, I'm not an expert with node or anything like that, I think I'm missing something because I can't seem to get the app running. Can I get a more detailed guide? Thanks!

**Comment by JuanCruz98** [comment]
What I did was:

- Created a new folder
- npm install @material/web
- Created the index.js file with the imports
- Created the index.html file
- Tried to open the index.html file

**Reply by asyncliz** [reply]
Have you tried the [build steps](https://github.com/material-components/material-web/blob/main/docs/quick-start.md#building) to bundle your JS file with rollup?

**Reply by JuanCruz98** [reply]
I tried now, the bundle.js file is the same as the index.js file, when trying to open the index.html file I get this error in the console:

Uncaught TypeError: Failed to resolve module specifier "@material/web/button/filled-button.js". Relative references must start with either "/", "./", or "../".

**Reply by asyncliz** [reply]
Hm, it doesn't sound like it was bundled properly. Can you share your rollup config?

**Comment by kavishdevar** [comment]
A few steps you could try that work for me..
- Install `rollup`
- Install `@rollup/plugin-node-resolve`
- Run `npx rollup -p @rollup/plugin-node-resolve index.js -o bundle.js`

_Demo index.js_
```js
import '@material/web/button/filled-button.js';
```

_Demo index.html_
```html
<head>
<script src='/bundle.js'></script>
</head>
<body>
<md-filled-button>label<md-filled-button>
</body>
```

I've also hosted a static js file I import in my other  projects on GitHub. I've included all available web-components javascript files required and bundled them.

_p.s. it also includes all components from [@mailoc07/material-web-additions](https://github.com/maicol07/material-web-additions)_

 You can find the static file here on my repo. [@stiblook/material-web-static](https://github.com/stiblook/material-web-static)
You can either download the js file from the repo, or use it directly in your HTML from [here](https://www.stiblook.rocks/material-web-static/material.js).

```html
<head>
<script src="https://www.stiblook.rocks/material-web-static/material.js"></script>
</head>
```
EDIT: You can also use `<script src='https://unpkg.com/@material/web@1.0.0-pre.11/button/outlined-button.js?module'</script>`
Regards

**Reply by JuanCruz98** [reply]
Thanks, now it worked!

**Comment by GitHubreham** [comment]
I tried now, the bundle.js file is the same as the index.js file, when trying to open the index.html file I get this error in the console:

Uncaught TypeError: Failed to resolve module specifier "@material/web/button/filled-button.js". Relative references must start with either
You can either download the js file from the repo, or use it directly in your HTML from here.
